Updated contributing documentation and setup steps for developers (#765)
Qiskit/qiskit-ibm-runtime: CONTRIBUTING.md

https://qiskit.org/documentation/contributing_to_qiskit.html

## Contributing to qiskit-ibm-runtime
-----------------------------------

In addition to the general guidelines there are specific details for
contributing to qiskit-ibm-runtime, these are documented below.

### Initial Steps

> 1. Fork your own copy of qiskit-ibm-runtime repository, then clone it in your 
>    local.

Reviewer: ```suggestion
1. Fork your own copy of `qiskit-ibm-runtime`, then clone it locally.
```

> 2. Setup a virtualenv based on the requirements.txt and requirements-dev.txt 

Reviewer: ```suggestion
2. Setup a virtual environment based on the requirements.txt and requirements-dev.txt 
```

   files.

> 3. Create a new branch and then make your contribution on it.

Reviewer: ```suggestion
3. Create a new branch and then make your contributions on it.
```

### Pushing a new branch

Before pushing your contribution please ensure that:

> 1. The code follows the code style of the project. For convenience, you can 
>    check [Style guide](#style-guide)

Reviewer: ```suggestion
1. The code follows the code style of this repository. For convenience, you can 
   check [Style guide](#style-guide)
```

> 2. If it makes sense for your change that you have added new tests that cover 
>    the changes.

Reviewer: ```suggestion
2. If it makes sense, add tests that cover the new changes.
```

> 3. All tests passes. For convenience, you can verify the [Test Types](#test).

Reviewer: ```suggestion
3. All tests pass. For convenience, you can verify the [Test Types](#test).
```

4. The documentation has been updated accordingly. In particular, if a
   function or class has been modified during your contribution, please update
   the *docstring* accordingly. For convenience, you can check [Building the 
   Documentation Locally](#building-documentation-locally).

### Pull request creation

When submitting a pull request and you feel it is ready for review,

> please ensure that if your change has an end user facing impact (new feature,
> deprecation, removal etc) that you have added a reno release note for that
> change and that the PR is tagged for the changelog.

Reviewer: ```suggestion
If your change has an end user facing impact (new feature,
deprecation, removal etc) please ensure that you add a reno release note and that the PR is tagged for the changelog.
```
